src/container.cpp

```diff
@@ -446,63 +444,63 @@ Cylinder* Container::queryDestination(int32_t& index, const Thing& thing, Item**
 		the slot position as if the bag has 20 slots
 		*/
 		index = INDEX_WHEREEVER;
-		*destItem = nullptr;
+		destItem = nullptr;
 	}
 
-	const Item* item = thing.getItem();
+	auto item = thing->getItem();
 	if (!item) {
-		return this;
+		return shared_from_this();
 	}
 
 	if (index != INDEX_WHEREEVER) {
-		Item* itemFromIndex = getItemByIndex(index);
+		auto itemFromIndex = getItemByIndex(index);
```

Comment:
Could this be `const auto`
```suggestion
		const auto itemFromIndex = getItemByIndex(index);
```